### DISCUSSION TITLE
How to use Dekorate outside of production code ?

### DISCUSSION THREAD

**Discussion by luistrigueiros** [dekorateio/dekorate#1036] [Ideas]
As a Java developer I found the the approach dekorate that attractive, but I would like to know if there is away to use Dekorate that we don't add the annotations directly in our production code, but into some other class or aspect that it is concerned with deployment of the application.
It does not looks clean to added deploy to kubernetes as concern into our production code.

**Comment by iocanel** [comment]
You can annotate any class.
It's even possible to not annotate at all in some cases (e.g. spring boot).

**Comment by luistrigueiros** [comment]
Are there any examples of this?

**Reply by aureamunoz** [reply]
Hello Luis,

Plenty of examples are available [here](https://github.com/dekorateio/dekorate/tree/main/examples). 
You can also find info about usage in the [doc](https://dekorate.io/docs/usage)

**Comment by luistrigueiros** [comment]
Hi @aureamunoz ,
Ok I got it but I mush confess I don't like it to edit spring application.properties it configure k8s yml deployment it is a full on regression.
The reason we have dekorate and is because we whant to have strong type checking and a better experience editing this than using the raw k8s yml in the firt place.
So the option I would looking for you to have is a way to define this annotions in another main other than in the main application production class as an additional ortogal concern.

**Reply by aureamunoz** [reply]
Ok, I haven't understood correctly.
As Ioannis mentioned, you can annotate any class and in some cases, using Spring Boot for example, just adding the dependency (no annotations at all), Dekorate will detect Spring Boot automatically and will generate a k8s manifest accordingly (with defaults values).

**Comment by luistrigueiros** [comment]
Ok so what I am looking to be able to express is have a multi-module project one with main production code and another module to drive the generation of k8s deployment yml's because for me deployment to kubernets does not belong in the main production code and to and dekorate annotions there is a nice to have not necessarily something one would desire to do.

**Reply by iocanel** [reply]
If using an independant module is a one way, then I am not sure it worths using dekorate. Dekorate is meant to be used per module, so you would end up having twice the amount for modules (1 for business and 1 for manifest generation).

**Comment by luistrigueiros** [comment]
So I like the idea of Dekorate, but honestly don't like the idea of having production code cluttered with Dekorate annotation, but also would not like to do this via a properties file as I would like to have a type safe API.
